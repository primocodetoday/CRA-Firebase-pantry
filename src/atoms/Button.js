import styled, { css } from 'styled-components/macro';
import PropTypes from 'prop-types';
import { defaultThemeProps } from 'theme/defaultThemeProps';

const Button = styled.button`
  background-color: ${({ theme }) => theme.secondary};
  padding: 9px 15px;
  font-size: ${({ theme }) => theme.fontSize.big};
  border: 2px solid white;
  color: white;
  border-radius: ${({ theme }) => theme.borderRadius};
  font-family: 'Montserrat', sans-serif;

  &:disabled,
  &[disabled] {
    border: 1px solid white;
    background-color: darkgrey;
    color: white;
  }

  ${({ secondary }) =>
    secondary &&
    css`
      background-color: grey;
    `}
`;

Button.propTypes = {
  secondary: PropTypes.bool,
  children: PropTypes.string.isRequired,
};

Button.defaultProps = defaultThemeProps;

export default Button;
