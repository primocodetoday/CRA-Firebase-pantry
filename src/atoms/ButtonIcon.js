import SC from 'styled-components/macro';
import { defaultThemeProps } from 'theme/defaultThemeProps';
import PropTypes from 'prop-types';

export const ButtonIcon = SC.button`
  width: 42px;
  height: 37px;
  background-image: url(${({ icon }) => icon});
  background-repeat: no-repeat;
  background-position: 50% 50%;
  background-size: contain;
  background-color: ${({ theme }) => theme.secondary};
  padding: 5px;
  border: 2px solid white;
  border-radius: ${({ theme }) => theme.borderRadius};
  cursor: pointer;
`;

ButtonIcon.propTypes = {
  icon: PropTypes.string,
};

ButtonIcon.defaultProps = defaultThemeProps;
