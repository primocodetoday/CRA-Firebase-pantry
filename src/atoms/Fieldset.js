import * as React from 'react';
import PropTypes from 'prop-types';
import SC from 'styled-components/macro';
import { defaultThemeProps } from 'theme/defaultThemeProps';

export const Wrapper = SC.fieldset`
  border-radius: ${({ theme }) => theme.borderRadius};
  padding: 25px;
  color: white;

  legend {
    font-size: ${({ theme }) => theme.fontSize.normal};
    padding: 0 10px;
    margin-left: 15px;
  }
`;

Wrapper.defaultProps = defaultThemeProps;

export const Fieldset = ({ legend, children, ...restProps }) => {
  const isLegendPassed = legend ? <legend>{legend}</legend> : null;

  return (
    <Wrapper {...restProps}>
      {isLegendPassed}
      {children}
    </Wrapper>
  );
};

Fieldset.defaultProps = {
  legend: '',
};

Fieldset.propTypes = {
  legend: PropTypes.string,
  children: PropTypes.node.isRequired,
};
