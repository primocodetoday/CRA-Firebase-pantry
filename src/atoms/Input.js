import styled, { css } from 'styled-components/macro';
import PropTypes from 'prop-types';
import { defaultThemeProps } from 'theme/defaultThemeProps';

const Input = styled.input`
  font-family: 'Montserrat', sans-serif;
  padding: 10px 10px;
  font-size: ${({ theme }) => theme.fontSize.normal};
  border-radius: ${({ theme }) => theme.borderRadius};
  border: none;
  color: ${({ theme }) => theme.secondary};
  text-align: center;

  &[type='number']::-webkit-inner-spin-button,
  &[type='number']::-webkit-outer-spin-button {
    margin: 0;
    -webkit-appearance: none;
  }

  &[type='number'] {
    -moz-appearance: textfield;
  }

  ${({ search }) =>
    search &&
    css`
      padding-left: 40px;
      width: 234px;
      height: 42px;
      color: ${({ theme }) => theme.secondary};
      font-size: ${({ theme }) => theme.fontSize.small};
      font-weight: 600;
      background-image: url('/assets/icons/search.svg');
      background-size: 20px;
      background-position: 10px 50%;
      background-repeat: no-repeat;
      text-align: left;
    `};

  ${({ settings }) =>
    settings &&
    css`
      text-align: center;
      color: ${({ theme }) => theme.secondary};
      font-weight: 600;
    `};
`;

Input.propTypes = {
  search: PropTypes.bool,
  settings: PropTypes.bool,
};

Input.defaultProps = defaultThemeProps;

export default Input;
