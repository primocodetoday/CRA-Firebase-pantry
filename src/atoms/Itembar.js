import styled from 'styled-components/macro';

const Itembar = styled.div`
  width: 80%;
  height: 1px;
  background-color: ${({ theme }) => theme.secondary};
`;

export default Itembar;
