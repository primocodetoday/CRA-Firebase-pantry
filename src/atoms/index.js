export { default as Button } from './Button';
export { default as ButtonIcon } from './ButtonIcon';
export { default as Fieldset } from './Fieldset';
export { default as Input } from './Input';
export { default as Logo } from './Logo';
export { default as Paragraph } from './Paragraph';
export { default as Section } from './Section';
export { default as Select } from './Select';
export { default as SignCard } from './SignCard';
export { default as SubTitle } from './Subtitle';
export { default as Title } from './Title';
export { default as MenuIcon } from './MenuIcon';
