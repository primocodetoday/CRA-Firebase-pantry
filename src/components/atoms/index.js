export { Button } from './Button/Button';
export { ButtonIcon } from './ButtonIcon/ButtonIcon';
export { Fieldset } from './Fieldset/Fieldset';
export { Input } from './Input/Input';
export { Logo } from './Logo/Logo';
export { Paragraph } from './Paragraph/Paragraph';
export { Section } from './Section/Section';
export { default as Select } from './Select/Select';
export { default as SubTitle } from './SubTitle/SubTitle';
export { default as Title } from './Title/Title';
export { default as SignCard } from './SignCard/SignCard';
