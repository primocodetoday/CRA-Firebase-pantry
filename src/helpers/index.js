export const filterIdById = (arr, id) => {
  const [item] = arr.filter((product) => product.id === id);
  return item ? item.id : null;
};

export const filterItemById = (arr, id) => {
  const [item] = arr.filter((product) => product.id === id);
  return item || null;
};

export const extractUnits = (arr) => {
  const units = [...new Set(arr.map((product) => product.unit))];
  return units || null;
};
