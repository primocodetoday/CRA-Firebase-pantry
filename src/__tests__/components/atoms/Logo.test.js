import React from 'react';
import { render } from '@testing-library/react';
import { TestThemeProvider } from '__tests__/helpers/TestThemeProvider';
import { Logo } from 'components/atoms';
import { MemoryRouter } from 'react-router-dom';

// TODO add routing testing

describe('Render <Logo/> properly', () => {
  it('with default props without provider', () => {
    const { getByText, container } = render(
      <MemoryRouter>
        <Logo />
      </MemoryRouter>,
    );

    expect(getByText('antry')).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('with default props with theme', () => {
    const { getByText, container } = render(
      <TestThemeProvider>
        <MemoryRouter>
          <Logo />
        </MemoryRouter>
      </TestThemeProvider>,
    );

    expect(getByText('antry')).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('with "big" props passed', () => {
    const { getByText, container } = render(
      <MemoryRouter>
        <Logo big />
      </MemoryRouter>,
    );

    expect(getByText('antry')).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
