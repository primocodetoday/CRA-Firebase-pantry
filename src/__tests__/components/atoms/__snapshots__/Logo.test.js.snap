// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render <Logo/> properly with "big" props passed 1`] = `
<a
  aria-current="page"
  class="StyledLogo__Wrapper-ssv8zj-0 bODzJp active"
  href="/"
  theme="[object Object]"
>
  <span
    class="StyledLogo__First-ssv8zj-1 fIwBVt"
  >
    P
  </span>
  antry
</a>
`;

exports[`Render <Logo/> properly with default props with theme 1`] = `
<a
  aria-current="page"
  class="StyledLogo__Wrapper-ssv8zj-0 ijTQGo active"
  href="/"
  theme="[object Object]"
>
  <span
    class="StyledLogo__First-ssv8zj-1 bvMZCy"
  >
    P
  </span>
  antry
</a>
`;

exports[`Render <Logo/> properly with default props without provider 1`] = `
<a
  aria-current="page"
  class="StyledLogo__Wrapper-ssv8zj-0 ijTQGo active"
  href="/"
  theme="[object Object]"
>
  <span
    class="StyledLogo__First-ssv8zj-1 fIwBVt"
  >
    P
  </span>
  antry
</a>
`;
